### PR TITLE
revert to import azure keyvault certs instead of generating them

### PR DESCRIPTION
**Pull request by ferantivero**
- modify narratives back to our previous strategy but keeping the
    current wording as much as we could
  - modify steps, arm and GitHub workflow o reinstate the base64 public cert

solved: #266614

### CODE REVIEW

File: 02-ca-certificates.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
+Now that you have the [prerequisites](./01-prerequisites) met, follow the steps below to create the TLS certificates that Azure Application Gateway will serve for clients connecting to your web app as well as the AKS Ingress Controller. If you already have access to an appropriate certificates, or can procure them from your organization, consider doing so and skipping the certificate generation steps. The following will describe using a self-signed certs for instructive purposes only.
 
 ## Steps
 
-1. Generate a self-signed TLS certificate
+1. Generate a client-facing self-signed TLS certificate
 
    > :book: Contoso Bicycle needs to procure a CA certificate for the web site. As this is going to be a user-facing site, they purchase an EV cert from their CA.  This will serve in front of the Azure Application Gateway.  They will also procure another one, a standard cert, to be used with the AKS Ingress Controller. This one is not EV, as it will not be user facing.
```
Comment (v-fearam) [inline 02-ca-certificates.md:10]: Need to be reviewed 
Line 13,
s, **a second TLS certificate is going to be generated, but that one will be generated directly from your Azure Key Vault.**

**Comment by v-fearam** [comment]
The 1-cluster-stamp.sh must be changed according to these changes.

File: 02-ca-certificates.md (hunk 1 above)
Comment (ferantivero) [reply 02-ca-certificates.md:10]: good catch

**Comment by v-fearam** [comment]
LGTM

File: 02-ca-certificates.md [hunk 2]
```diff
@@ -1,29 +1,45 @@
...
+
+1. Generate the wildcard certificate for the Ingress Controller using Azure KeyVault
+
+   > :book: Contoso Bicycle will also procure another TLS certificate, a standard cert, to be used with the AKS Ingress Controller. This one is not EV, as it will not be user facing. Finally the app team decides to use a wildcard certificate of `*.aks-ingress.contoso.com` for the ingress controller.
+
+   ```bash
+   openssl req -x509 -nodes -days 365 -newkey rsa:2048 -out traefik-ingress-internal-aks-ingress-contoso-com-tls.crt -keyout traefik-ingress-internal-aks-ingress-contoso-com-tls.key -subj "/CN=*.aks-ingress.contoso.com/O=Contoso Aks Ingress"
+   ```
+
+1. Base64 encode the AKS Ingress Controller TLS certificate
```
Comment (ckittel) [inline 02-ca-certificates.md:36]:
```suggestion
1. Base64 encode the AKS Ingress Controller certificate
```

To be consistent with the "client-facing" one.

File: 02-ca-certificates.md [hunk 3]
```diff
@@ -1,29 +1,45 @@
...
-   No matter if you used a certificate from your organization or you generated one from above, you'll need the certificate (as `.pfx`) to be base 64 encoded for proper storage in Key Vault later.
+   :bulb: No matter if you used a certificate from your organization or you generated one from above, you'll need the certificate (as `.pfx`) to be base 64 encoded for proper storage in Key Vault later.
 
    ```bash
-    export APP_GATEWAY_LISTENER_CERTIFICATE=$(cat appgw.pfx | base64 -w 0)
-    ```
+   export APP_GATEWAY_LISTENER_CERTIFICATE=$(cat appgw.pfx | base64 -w 0)
+   ```
+
+1. Generate the wildcard certificate for the Ingress Controller using Azure KeyVault
```
Comment (ckittel) [inline 02-ca-certificates.md:28]:
```suggestion
1. Generate the wildcard certificate for the AKS Ingress Controller
```

File: 05-aks-cluster.md [hunk 4]
```diff
@@ -80,6 +80,18 @@ Now that the [hub-spoke network is provisioned](./04-networking.md), the next st
        echo $APP_GATEWAY_LISTENER_CERTIFICATE
        ```
 
+    1. Create `AKS_INGRESS_CONTROLLER_CERTIFICATE_BASE64` secret in your GitHub repository. For more
+       information, please take a look at [Creating encrypted secrets for a repository](https://docs.github.com/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository).
```
Comment (ckittel) [inline 05-aks-cluster.md:84]:
```suggestion
    1. Create `AKS_INGRESS_CONTROLLER_CERTIFICATE_BASE64` secret in your GitHub repository. For more information, please take a look at [Creating encrypted secrets for a repository](https://docs.github.com/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository).
```

File: 05-aks-cluster.md [hunk 5]
```diff
@@ -80,6 +80,18 @@ Now that the [hub-spoke network is provisioned](./04-networking.md), the next st
...
+
+       > :bulb:
+       >
+       >  * Use the env var value of `AKS_INGRESS_CONTROLLER_CERTIFICATE_BASE64`
+       >  * Ideally fetching this secret from a platform-managed secret store such as [Azure KeyVault](https://github.com/marketplace/actions/azure-key-vault-get-secrets)
```
Comment (ckittel) [inline 05-aks-cluster.md:89]:
```suggestion
       >  * Ideally fetching this secret from a platform-managed secret store such as [Azure Key Vault](https://github.com/marketplace/actions/azure-key-vault-get-secrets)
```

File: 07-workload-prerequisites.md [hunk 6]
```diff
@@ -1,91 +1,31 @@
 # Workflow Prerequisites
 
-The AKS Cluster has been enrolled in [GitOps management](./06-gitops), wrapping up the infrastructure focus of the [AKS secure Baseline reference implementation](./). Follow the steps below to create the TLS certificate that the Ingress Controller will serve for Application Gateway to connect to your web app.
+The AKS Cluster has been enrolled in [GitOps management](./06-gitops), wrapping up the infrastructure focus of the [AKS secure Baseline reference implementation](./). Follow the steps below to import the TLS certificate that the Ingress Controller will serve for Application Gateway to connect to your web app.
 
 ## Steps
 
-## Generate the wildcard certificate for the Ingress Controller using Azure KeyVault
+## Import the wildcard certificate for the Ingress Controller to Azure KeyVault
```
Comment (ckittel) [inline 07-workload-prerequisites.md:7]:
```suggestion
## Import the wildcard certificate for the AKS Ingress Controller to Azure Key Vault
```

File: 07-workload-prerequisites.md [hunk 7]
```diff
@@ -1,91 +1,31 @@
...
+   > :book: Finally the app team decides to use a wildcard certificate of `*.aks-ingress.contoso.com` for the ingress controller. They use Azure Key Vault to import and manage the lifecycle of this certificate.
 
    ```bash
    KEYVAULT_NAME=$(az deployment group show --resource-group rg-bu0001a0008 -n cluster-stamp --query properties.outputs.keyVaultName.value -o tsv)
-   az keyvault set-policy --certificate-permissions create list get -n $KEYVAULT_NAME --upn $(az account show --query user.name -o tsv)
+   az keyvault set-policy --certificate-permissions import list get -n $KEYVAULT_NAME --upn $(az account show --query user.name -o tsv)
    ```
 
-1. Generate the Cluster Ingress Controller's Wildcard Certificate for `*.aks-ingress.contoso.com`.
+1. Import the Cluster Ingress Controller's Wildcard Certificate for `*.aks-ingress.contoso.com`.
```
Comment (ckittel) [inline 07-workload-prerequisites.md:20]:
```suggestion
1. Import the AKS Ingress Controller's Wildcard Certificate for `*.aks-ingress.contoso.com`.
```

File: 07-workload-prerequisites.md [hunk 8]
```diff
@@ -1,91 +1,31 @@
...
 
-    :warning: If you already have access to an appropriate certificate, or can procure one from your organization, consider skipping this step and instead [import it into Key Vault](https://docs.microsoft.com/azure/key-vault/certificates/tutorial-import-certificate#import-a-certificate-to-key-vault).
+   :warning: If you already have access to an appropriate certificate, or can procure one from your organization, consider using it from this very same step. For more information, please take a look at the [import certificate tutorial using Azure Key Vault](https://docs.microsoft.com/azure/key-vault/certificates/tutorial-import-certificate#import-a-certificate-to-key-vault).
```
Comment (ckittel) [inline 07-workload-prerequisites.md:22]:
```suggestion
   :warning: If you already have access to an appropriate certificate, or can procure one from your organization, consider using it for this step. For more information, please take a look at the [import certificate tutorial using Azure Key Vault](https://docs.microsoft.com/azure/key-vault/certificates/tutorial-import-certificate#import-a-certificate-to-key-vault).
```

File: README.md [hunk 9]
```diff
@@ -54,7 +54,7 @@ A deployment of AKS-hosted workloads typically experiences a separation of dutie
 There are considerations that must be addressed before you start deploying your cluster. Do I have enough permissions in my subscription and AD tenant to do a deployment of this size? How much of this will be handled by my team directly vs having another team be responsible?
 
 * [ ] Begin by ensuring you [install and meet the prerequisites](./01-prerequisites.md)
-* [ ] [Procure client-facing TLS certificate](./02-ca-certificates.md)
+* [ ] [Procure client-facing and the AKS Ingress Controller TLS certificates](./02-ca-certificates.md)
```
Comment (ckittel) [inline README.md:57]:
```suggestion
* [ ] [Procure client-facing and AKS Ingress Controller TLS certificates](./02-ca-certificates.md)
```

**Pull request merged**